### PR TITLE
add interfaces to provide marshal/unmarshal logic for unknown fields in struct v2

### PR DESCRIPTION
**Pull request by skinass**

### CODE REVIEW

File: helpers.go [hunk 1]
```diff
@@ -26,6 +26,16 @@ type Optional interface {
...
 }
 
+// UnknownsUnmarshaler provides a method to unmarshal unknown struct fileds and save them as you want
+type UnknownsUnmarshaler interface {
+	UnmarshalUnknown(key string, value interface{})
+}
+
+// UnknownsMarshaler provides a method to write additional struct fields
+type UnknownsMarshaler interface {
+	MarshalUnknowns() map[string]interface{}
```
Comment (vstarodub) [inline helpers.go]: I don't particularly like having a map return value.

IMO, it's better to have `MarshalUnknowns(w *jwriter.Writer, empty bool)`. This way, using a map (or even interface-es) will not be required. Also this allows for more use cases.

File: helpers.go [hunk 2]
```diff
@@ -26,6 +26,16 @@ type Optional interface {
 	IsDefined() bool
 }
 
+// UnknownsUnmarshaler provides a method to unmarshal unknown struct fileds and save them as you want
+type UnknownsUnmarshaler interface {
+	UnmarshalUnknown(key string, value interface{})
```
Comment (vstarodub) [inline helpers.go]: It's better to allow returning an error, it may be useful for custom validation/unmarshalling.

File: helpers.go (hunk 1 above)
Comment (skinass) [reply helpers.go]: then check the first version of this PR here: https://github.com/mailru/easyjson/pull/267/files

File: helpers.go (hunk 2 above)
Comment (skinass) [reply helpers.go]: in this PR https://github.com/mailru/easyjson/pull/267/files i changed this interface to 
```go
type UnknownsUnmarshaler interface {
	UnmarshalUnknown(in *jlexer.Lexer, key string)
}
```

so developers can use jlexer.Lexer to put errors

**Pull request closed without merging**